content/ui-patterns/notification-messaging.mdx

```diff
@@ -0,0 +1,174 @@
+---
+
+import { Box } from '@primer/react'
+import { InfoIcon, CheckCircleIcon, StopIcon, AlertIcon } from '@primer/octicons-react'
+
+Primer messaging components:
+
+- Announcement (not yet available)
+- [Banner](../components/banner)
+- Inline message (coming soon)
+
+## Message types
+
+Messaging holds differing levels of prominence which can be used to determine the appropriate component to use. Most messages fall into one of the following categories, from most prominent to least:
+
+**System updates**: These types of message originate from GitHub and are not user initiated. A global site wide announcement (outage, missing payment details, critical and blocking) or messaging related to user account/permissions should use the **Announcement** component. System announcements related to a specific product area such as a single sign-on message should use the **Banner** component. System update type messages cannot be dismissed until the issue is resolved, or may be perminent if they are purely informative.
+
+**Feedback**: Communicates the result of a user action (submitting a form, toggling a setting, etc.) Feedback can be positive, negative, or a warning. Feedback is shown in either a **Banner** or **Inline message**. For forms, an inline message is used for individual field validation which may be combined with a banner if multiple fields have errors. Banners are placed at the top of the page or section they are related to, but still within the body content. Inline messages are placed near the action they are related to.
+
+**Awareness**: Communicates information that is relevant to the user but not necessarily related to an action they've taken. This type of message is typically used to provide context or additional information about a feature or product. A **Banner** using the `info` state might be used to suggest an action, while an **InlineMessage** might suggest a helpful tip.
+
+## Message states
+
+<img
+  width="960"
+  alt="A group of labels showing the proper icon and color combination for each message state"
+  src="https://github.com/primer/design/assets/18661030/bbcbe8a0-4d7f-47f0-9f20-c6cee406115f"
+/>
+
+### Visual treatment
+
+Primer offers six states for messaging components: info, warning, success, unavailable, critical, and upsell. Each state has a corresponding icon and color combination to help communicate the message's intent.
+
+Icons sized at 12px utilize the filled variant, white 14px and above utilize the outline variant.
+
+<img
+  width="960"
+  alt="An example of an input field with a 12px error message next to two examples of 14px icon usage, which uses an outline icon."
+  src="https://github.com/primer/design/assets/18661030/6ab6f9e9-d7e9-4f75-907f-0e08aec0db3c"
+/>
+
+### Overview
+
+Messaging components combine state with message type to provide the appropriate level of prominence. To determine which component to use for your message, start by identifying the message state.
```

Comment:
I would love to see a Do/Don't section to demonstrate any improper uses of components we've found and what should be done instead!!